corva/models/base.py

```diff
@@ -1,47 +1,87 @@
 from __future__ import annotations
 
 from abc import ABC, abstractmethod
-from typing import Any, List, Optional, TypeVar
+from typing import Any, Generic, List, Optional, Type, TypeVar, Union
 
 from pydantic import BaseModel, Extra
+from pydantic.generics import GenericModel
 
 from corva.network.api import Api
+from corva.settings import Settings
+from corva.state.redis_adapter import RedisAdapter
 from corva.state.redis_state import RedisState
 
 
+class BaseConfig:
+    allow_population_by_field_name = True
+    arbitrary_types_allowed = True
+    extra = Extra.allow
+    validate_assignment = True
+
+
 class BaseEvent(ABC):
     @staticmethod
     @abstractmethod
-    def from_raw_event(event: str, **kwargs) -> BaseEvent:
+    def from_raw_event(event: str, **kwargs) -> Union[List[BaseEvent], BaseEvent]:
+        pass
+
+
+class BaseData(BaseModel):
+    class Config(BaseConfig):
```

Comment:
is there any chance we can void using nested classes?

Reply:
we can do
```suggestion
    class BaseData(BaseModel):
        Config = BaseConfig
```

Reply:
resolved the issue using base classes [CorvaBaseModel](https://github.com/corva-ai/python-sdk/blob/911358315edce6151fbbb48faea021a7460488db/corva/models/base.py#L29)